src/hostcalls.rs

```diff
@@ -21,12 +21,20 @@ extern "C" {
     fn proxy_log(level: LogLevel, message_data: *const u8, message_size: usize) -> Status;
 }
 
+// Useful for the panic hook to avoid a double-panic.
+#[inline]
+pub(crate) fn log_nopanic(level: LogLevel, message: &str) -> Result<Status, Status> {
```

Comment:
Is there any reason for using `Result<Status, Status>` instead of `Result<(), Status>` like rest of the SDK?

Reply:
Just leaving the decision about what to do with the `Status` to the callers out of habit. Since this fn is private to this crate it won't show as part of the SDK to users, and this allows for callers to have more flexibility to make that decision or be extended based on this value without needing to modify this fn.